---
title: "Parkinson's Disease RNA-Seq analysis"
author: "Gabriel Ramirez-Vilchis"
date: "`r Sys.Date()`"
output: 
  html_document:
    toc: yes
    theme: united
    number_sections: true
    highlight: tango
toc-title: "Table of contents"
bibliography: ./references.bib
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Introduction

Parkinson's disease (PD) is a neurodegenerative disorder. It consists of 
neuronal loss in the substantia nigra, which causes striatal dopamine 
deficiency. Important hallmarks of PD are aggregates of α-synuclein 
(α-syn). In this project, RNA-seq data from mice with induced α-syn 
accumulation is analyzed in order to learn more about microglia role in PD 
[@Poewe2017].  

# Antecedents and data source  

It is known that activated microglia is usually present in PD and it can 
promote α-syn aggregation. Nevertheless, it is not yet completely 
understood how microglia participates in α-syn cell-to-cell transfer.  

In order to study so, a research group analyzed α-syn propagation between 
cells in a mouse model in conditions that activated or depleted microglial 
cells [@George2019]. They stimulated microglia with either lipopolysaccharide 
(LPS) or with interleukin-4 (IL-4), and used PBS as a control. By doing so, 
they generated situations of depleted (PBS), low-activated (IL-4) and 
high-activated (IL-4) microglia.  

I analyzed data generated during the research previously mentioned. The data 
is available at <https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE130683>.

# Hypothesis  

Stimulation of microglia affects α-syn transfer between neuronal cells.  

# Objective  

To analyze RNA-seq data from mice with induced α-syn accumulation in order to 
determine the effect of microglia stimulation on α-syn transfer between cells.

# Data processing

## Library importation

First, the libraries needed for the analysis are imported. They include libraries for data manipulation, normalization, and visualization. 

```{r libraries, message=FALSE}
# Import libraries
library(recount3)
library(SummarizedExperiment)
library(edgeR)
library(ggplot2)
library(limma)
library(pheatmap)
library(RColorBrewer)
```

## Data selection  

The data is selected from the available projects in _recount3_. The project 
selected is SRP194918, which contains data from mice with induced α-syn 
accumulation.

```{r project_selection}
# List all available projects for mouse
mouse_projects <- available_projects(organism = "mouse")

# Extract the project information
proj_info <- subset(
  mouse_projects,
  project == "SRP194918" & project_type == "data_sources"
)

# Create a recount object for the SRP058181 project
rse_gene_SRP194918 <- create_rse(proj_info)
rse_gene_SRP194918

# Analyze the project
rowData(rse_gene_SRP194918)
rowRanges(rse_gene_SRP194918)

```

## Data formatting

The data is transformed to read counts and important data for analysis is 
added. The data types are converted to factors and the data is summarized.  

```{r formatting}
# Transform nucleotide counts to read counts
assay(rse_gene_SRP194918, "counts") <- compute_read_counts(rse_gene_SRP194918)

# Add important data for analysis
rse_gene_SRP194918 <- expand_sra_attributes(rse_gene_SRP194918)

# Check the data format
rse_gene_SRP194918$sra.sample_attributes[1:3]

# Review data of interest
colData(rse_gene_SRP194918)[
  ,
  grepl("^sra_attribute", colnames(colData(rse_gene_SRP194918)))
]

# Convert data types to factors
rse_gene_SRP194918$sra_attribute.age <-
  factor(rse_gene_SRP194918$sra_attribute.age)
rse_gene_SRP194918$sra_attribute.injected_with <-
  factor(rse_gene_SRP194918$sra_attribute.injected_with)
rse_gene_SRP194918$sra_attribute.source_name <-
  factor(rse_gene_SRP194918$sra_attribute.source_name)
rse_gene_SRP194918$sra_attribute.strain <-
  factor(rse_gene_SRP194918$sra_attribute.strain)
rse_gene_SRP194918$sra_attribute.tissue <-
  factor(rse_gene_SRP194918$sra_attribute.tissue)
rse_gene_SRP194918$sra_attribute.treatment <-
  factor(rse_gene_SRP194918$sra_attribute.treatment)

# Summarize data
summary(as.data.frame(colData(rse_gene_SRP194918)[
  ,
  grepl("^sra_attribute", colnames(colData(rse_gene_SRP194918)))
]))
```

There are 17 samples. While the only relevant difference is the injection, I 
analyzed the effects that it has in gene expression.  

## Data filtering

The data is filtered to remove genes with low expression.  

```{r filtering}
# Check quality of the data
rse_gene_SRP194918$assigned_gene_prop <-
  rse_gene_SRP194918$recount_qc.gene_fc_count_all.assigned /
  rse_gene_SRP194918$recount_qc.gene_fc_count_all.total
summary(rse_gene_SRP194918$assigned_gene_prop)

# Observe differences among samples injected with different substances
with(colData(rse_gene_SRP194918),
     tapply(assigned_gene_prop,
            sra_attribute.injected_with,
            summary))

# Plot the distribution of the assigned gene proportion
hist(rse_gene_SRP194918$assigned_gene_prop, 
     col = "#fd8d3c", 
     main = "Assigned Gene Proportion", 
     xlab = "Assigned Gene Proportion")

# Save the unfiltered data
rse_gene_SRP194918_unfiltered <- rse_gene_SRP194918

# Filter genes with low expression
rse_gene_SRP194918 <-
  rse_gene_SRP194918[edgeR::filterByExpr(rse_gene_SRP194918),]

# Plot the distribution of the assigned gene proportion after filtering
hist(rse_gene_SRP194918$assigned_gene_prop, 
     col = "#fd8d3c", 
     main = "Filtered Assigned Gene Proportion",
     xlab = "Assigned Gene Proportion")

# Check the expression of genes that were not filtered out
gene_means <- rowMeans(assay(rse_gene_SRP194918, "counts"))
summary(gene_means)

# Check the percentage of genes that were not filtered out
round(nrow(rse_gene_SRP194918) / nrow(rse_gene_SRP194918_unfiltered) * 100, 2)
```

## Data normalization

The data is normalized using Log-fold changes.  

```{r normalization}
# Create the list for normalization
dge <- DGEList(counts = assay(rse_gene_SRP194918, "counts"),
               genes = rowData(rse_gene_SRP194918))

# Calculate the normalization factors
dge <- calcNormFactors(dge)
```

# Differential expression analysis

The differential expression analysis is performed using the _limma_ package.  

Assigned gene proportion was plotted according to the injection group, showing 
considerable differences among them.  

Model matrix was created according to the injection group and the assigned 
gene proportion.  

Most differentially expressed genes were calculated using the Bayes method and 
were identified in a Volcano plot. Also a heatmap was created with the 50 top 
genes to analyze differential expression.  

```{r analysis}
# Plot gene expression according to the injection group
ggplot(as.data.frame(colData(rse_gene_SRP194918)),
       aes(y = assigned_gene_prop,
           x = sra_attribute.injected_with)) +
  geom_boxplot() +
  ylab("Assigned Gene Prop") +
  xlab("Injection Group") +
  theme_classic()

# Create the model matrix according to the injection group
mod <- model.matrix(~ sra_attribute.injected_with + assigned_gene_prop,
                    data = colData(rse_gene_SRP194918)
)
colnames(mod)
# Estimate the mean-variance relationship
vGene <- voom(dge, mod, plot = TRUE)

# Compute statistics by Bayes method
eb_results <- eBayes(lmFit(vGene))

# Extract the top genes
de_results <- topTable(
  eb_results,
  coef = c(2, 3),
  number = nrow(rse_gene_SRP194918),
  sort.by = "none"
)

# Calculate the number of significant genes
table(de_results$adj.P.Val < 0.05)

# Plot the differentially expresed genes when injected with LPS
volcanoplot(eb_results, coef = 2, highlight = 3, names = de_results$gene_name)

# Review the top genes information
de_results[de_results$gene_name %in% c("Jchain", "Lox", "Slc2a5"), ]

# Heatmap of the 50 top genes
exprs_heatmap <- vGene$E[rank(de_results$adj.P.Val) <= 50, ]

# Create a dataframe with the injection group
df <- as.data.frame(colData(rse_gene_SRP194918)$sra_attribute.injected_with)

# Rename the columns and rows
colnames(df) <- c("Injection")
rownames(df) <- colnames(exprs_heatmap)

# Change the gene ID for the gene name
row.names(exprs_heatmap) <-
  de_results$gene_name[rank(de_results$adj.P.Val) <= 50]

# Create the heatmap
pheatmap(
  exprs_heatmap,
  cluster_rows = TRUE,
  cluster_cols = TRUE,
  show_colnames = FALSE,
  annotation_col = df,
  scale = "row",
)

# MDS plot according to the injection group
col.group <- df$Injection
levels(col.group) <- brewer.pal(nlevels(col.group), "Set1")
col.group <- as.character(col.group)
plotMDS(vGene$E, labels = df$Injection, col = col.group)

```

# Biological discussion and conclusions  

In the assignment gene proportion boxplot, it is observed that the group 
injected with LPS has a median significantly different to other groups. This 
suggests that the injection with LPS has a significant effect on gene 
expression, and this is because LPS was a potent activator of microglia.  

The differential expression analysis showed that the group injected with LPS 
was slightly more affected than the group injected with PBS. This is because 
LPS is a potent activator of microglia, and it is known that activated 
microglia can promote α-syn aggregation. It is imortant to mention that model 
was built using coefficients 2 and 3, because those were the columns 
indicating injections with either PBS or LPS (if none of them, that means that 
the sample was injected with IL-4) Also, the 3 top genes that were most 
differentially expressed in the group injected with LPS were Jchain, Lox, and 
Slc2a5. These genes are related to the immune response, oxidative stress, and 
glucose transport, respectively; and they are also involved in inflammation, 
which is a process that is usually present in PD.  

Finally, the heatmap showed that the group injected with LPS had a different 
expression pattern compared to the group injected with PBS and IL-4. This 
suggests that the injection with LPS has a significant effect on expression of 
many genes related to inflammation and oxidative stress. The MDS plot also 
showed a clear separation between the group injected with LPS and the other 
groups.  

With all of this, it could be concluded that activation of microglia does have 
an effect on α-syn transfer between cells, and that it could be a potential 
target for PD treatment.

# References
